docs/existing-aws-infrastructure.md

```diff
@@ -0,0 +1,94 @@
+# Consuming Existing AWS Infrastructure
+
+Normally, Cluster API will create infrastructure on AWS when standing up a new workload cluster. However, it is possible to have Cluster API re-use existing AWS infrastructure instead of creating its own infrastructure. Follow the instructions below to configure Cluster API to consume existing AWS infrastructure.
+
+## Prerequisites
+
+In order to have Cluster API consume existing AWS infrastructure, you will need to have already created the following resources:
+
+* A VPC
+* One or more private subnets (subnets that do not automatically assign a public IP address to EC2 instances)
+* A public subnet in the same Availability Zone (AZ) for each private subnet (this is required for NAT gateways to function properly)
+* A NAT gateway for each private subnet, along with associated Elastic IP addresses
+* An Internet gateway for all public subnets
+* Route table associations that provide connectivity to the Internet through a NAT gateway (for private subnets) or the Internet gateway (for public subnets)
```

Comment:
It might be worth mentioning that that the public subnet / IGW bits are still necessary, even if you also specify an `internal` scheme for your [apiserver's load balancer](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/api/v1alpha3/types.go#L121). It looks like we're [unwilling to proceed without one](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0b83f06314410a9a3d628f006a647086f51736a8/pkg/cloud/services/ec2/subnets.go#L67).

Reply:
Actually, we do [defer to the public role tag](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0b83f06314410a9a3d628f006a647086f51736a8/pkg/cloud/services/ec2/subnets.go#L210-L212), so if someone _really_ wants to they could tag an actually-private subnet as role "public" to get past that linked error.

Probably best to leave that out of the documentation, though, as I've no idea what might fail down the line if someone did such a thing.

Reply:
Updated via 38ce860. I did not add information specific to the requirement of a public subnet and IGW with regard to the API server load balancer, as those items are already needed anyway. If desired, I can add those details.

Reply:
I didn't specifically address this comment (the text already noted that the public subnet and IGW are required), but I can add some additional text if desired.